$OpenBSD: patch-src_hooks_c,v 1.3 2011/10/01 14:31:04 kurt Exp $

Write output immediately in case user is interacting with a pipeline.

--- src/hooks.c.orig	Thu Dec 31 13:40:47 2009
+++ src/hooks.c	Thu Sep  8 20:33:02 2011
@@ -29,8 +29,9 @@ void jam_fprintf(FILE *stream, const char *fmt, ...) {
 
     va_start(ap, fmt);
     (*vfprintf_hook)(stream, fmt, ap);
-
     va_end(ap);
+
+    fflush(stream);
 }
 
 void jamvm_exit(int status) {
